File: frontend/src/types/source.ts
export interface VoiceActionItem {
  text: string;
  context?: string;
}

export interface VoiceStructuredNote {
  summary: string;
  key_points: string[];
  action_items: VoiceActionItem[];
  ideas: string[];
  open_questions: string[];
}

export interface Source {
  id: string;
  title: string;
  content: string;
  source_type: string;
  meta_info: {
    media?: {
      storage_path?: string;
      mime_type?: string;
      media_type?: string;
      original_filename?: string;
      structured_note?: VoiceStructuredNote;
      smart_chapters?: any[];
      transcript_segments?: Array<{
        start: number;
        end: number;
        text: string;
      }>;
    };
    transcription?: {
      status?: string;
      latency_sec?: number;
    };
    transcript_segments?: Array<{
      start: number;
      end: number;
      text: string;
    }>;
    insights?: Record<string, any>;
    [key: string]: any;
  };

  // Source Manager 2.0 fields
  file_type?: string;
  original_file_path?: string;
  raw_content?: string;
  domain?: string;
  version: number;
  is_deleted: boolean;
  metadata_info: Record<string, any>;

  status: string;
  error_message?: string;
  started_at?: string;
  completed_at?: string;
  created_at: string;
  updated_at: string;

  chunks_count: number;
  claims_count: number;
}

export interface SourceCreate {
  title: string;
  content: string;
  source_type?: string;
  meta_info?: Record<string, any>;
}

export interface SourceDetail extends Source {
  chunks: Array<{
    id: string;
    chunk_index: number;
    text_content: string;
  }>;
  claims: Array<{
    id: string;
    content: string;
    claim_type: string;
    category: string;
    confidence: number;
    is_active: boolean;
    superseded_by: string | null;
  }>;
}

File: backend/app/schemas/source.py
import uuid
from datetime import datetime
from typing import Any, Dict, List, Optional
from pydantic import BaseModel, Field


class SourceBase(BaseModel):
    title: str
    domain: Optional[str] = None
    importance: str = "normal"
    folder: Optional[str] = None


class SourceCreate(SourceBase):
    content: str
    source_type: str = "note"
    meta_info: Dict[str, Any] = Field(default_factory=dict)


class SourceUpdateContent(BaseModel):
    raw_content: str
    domain: Optional[str] = None


class SourceUpdateFolder(BaseModel):
    folder: Optional[str] = None


class ChunkSimple(BaseModel):
    id: str
    chunk_index: int
    text_content: str


class ClaimSimple(BaseModel):
    id: str
    content: str
    claim_type: str
    category: Optional[str] = None
    confidence: float
    is_active: bool
    superseded_by: Optional[str] = None


class SourceResponse(BaseModel):
    id: uuid.UUID
    title: str
    content: Optional[str] = None
    source_type: str
    meta_info: Dict[str, Any] = Field(default_factory=dict)
    file_type: Optional[str] = None
    original_file_path: Optional[str] = None
    raw_content: Optional[str] = None
    domain: Optional[str] = None
    folder: Optional[str] = None
    version: int
    is_deleted: bool
    metadata_info: Dict[str, Any] = Field(default_factory=dict)
    status: str
    error_message: Optional[str] = None
    started_at: Optional[datetime] = None
    completed_at: Optional[datetime] = None
    created_at: datetime
    updated_at: datetime
    chunks_count: int = 0
    claims_count: int = 0

    class Config:
        from_attributes = True


class SourceDetailResponse(SourceResponse):
    chunks: List[ChunkSimple] = []
    claims: List[ClaimSimple] = []
